alns/ALNS.py

```diff
@@ -109,10 +119,63 @@ def add_repair_operator(self, op: _OperatorType, name: str = None):
         name
             Optional name argument, naming the operator. When not passed, the
             function name is used instead.
+        only_after
+            Optional keyword-only argument to indicate after which destroy
+            operators the passed-in repair operator can be used. If None,
+            then it is assumed that all destroy operators can be used.
         """
         logger.debug(f"Adding repair operator {op.__name__}.")
         self._repair_operators[name if name else op.__name__] = op
 
+        if only_after is not None:
+            self._only_after[op].update(only_after)
+
+    def _compute_op_coupling(self) -> np.ndarray:
+        """
+        Internal helper to compute a matrix that describes the
+        coupling between destroy and repair operators. The matrix has size
+        |d_ops| x |r_ops| and entry (i, j) is 1 if destroy operator i can
+        be used in conjunction with repair operator j and 0 otherwise.
+
+        If the only_after keyword argument was not used in when
+        adding repair operators, then all entries of the matrix are 1.
+        """
+        op_coupling = np.zeros(
+            (len(self.destroy_operators), len(self.repair_operators))
+        )
+
+        for r_idx, (_, r_op) in enumerate(self.repair_operators):
+            only_after = self._only_after[r_op]
+
+            if len(only_after) == 0:
+                # Couple all destroy operators if only_after is unspecified
+                op_coupling[:, r_idx] = 1
+
+            else:
+                for d_idx, (_, d_op) in enumerate(self.destroy_operators):
+                    if d_op in only_after:
+                        op_coupling[d_idx, r_idx] = 1
+
+        # Operator coupling can only be computed for added destroy operators
+        for only_after in self._only_after.values():
+            for d_op in only_after:
+                if d_op not in self._destroy_operators.values():
+                    raise ValueError(
+                        f"{d_op.__name__} was passed in the only_after\
+                        argument when adding repair operators but has not been\
+                        added as destroy operator."
+                    )
```

Comment:
Question: if passed an unused operator, what happens then? What goes wrong? 

As far as I can tell, that unused only-after operator is ignored, which is not necessarily something to throw an error for. It is mostly *weird*, but not problematic.

Reply:
Nothing goes wrong indeed! I think I was being pedantic when writing out this test. I'll remove it.